srtcore/epoll.cpp

```diff
@@ -273,34 +230,19 @@ int CEPoll::update_usock(const int eid, const SRTSOCKET& u, const int* events)
    if (p == m_mPolls.end())
       throw CUDTException(MJ_NOTSUP, MN_EIDINVAL);
 
-   if (!events || (*events & UDT_EPOLL_IN))
-      p->second.m_sUDTSocksIn.insert(u);
-   else
+  int evts = events ? *events : (UDT_EPOLL_IN | UDT_EPOLL_OUT | UDT_EPOLL_ERR);
+	if(evts)
    {
-      p->second.m_sUDTSocksIn.erase(u);
-      /*
-      * We are no longer interested in this event from this socket
-      * If some are up, they will unblock EPoll forever.
-      * Clear them.
-      */
-      p->second.m_sUDTReads.erase(u);
+		pair<int, int>& wait = p->second.m_sUDTSocksWait[u];
+		wait.first = evts; // watching
+		evts = wait.first & wait.second; // watching & state!
+		if(evts) {
+			p->second.m_sUDTSocksSet[u] = evts;
+			return 0;
    }
-
-   if (!events || (*events & UDT_EPOLL_OUT))
-      p->second.m_sUDTSocksOut.insert(u);
-   else
-   {
-      p->second.m_sUDTSocksOut.erase(u);
-      p->second.m_sUDTWrites.erase(u);
-   }
-   if (!events || (*events & UDT_EPOLL_ERR))
-      p->second.m_sUDTSocksEx.insert(u);
-   else
-   {
-      p->second.m_sUDTSocksEx.erase(u);
-      p->second.m_sUDTExcepts.erase(u);
-   }
-
+	} else 
+		p->second.m_sUDTSocksWait.erase(u);
+	p->second.m_sUDTSocksSet.erase(u);
```

Comment:
Please use space-based alignment like in the rest of the file, instead of the tab-based.